bundles/io/org.eclipse.smarthome.io.transport.mdns/src/main/java/org/eclipse/smarthome/io/transport/mdns/discovery/MDNSDiscoveryService.java

```diff
@@ -68,59 +73,41 @@ public void run() {
                 startScan();
             }
         }, 0, TimeUnit.SECONDS);
-        for (MDNSDiscoveryParticipant participant : participants) {
-            mdnsClient.getClient().addServiceListener(participant.getServiceType(), this);
-        }
-    }
-
-    @Override
-    protected void stopBackgroundDiscovery() {
-        for (MDNSDiscoveryParticipant participant : participants) {
```

Comment:
Ok, then let me ask you differently: Why did you remove the code within this method? If no background discovery is wished, no listeners should be registered and no devices should be found.